frontend/src/widgets/importancecomponent.js

```diff
@@ -24,81 +24,69 @@ import { Settings } from '../settings';
 import WidgetHeader from '../components/widget-header';
 import ParamDropdown from '../components/param-dropdown';
 
-export class ImportanceComponentWidget extends React.Component {
-  static propTypes = {
-    title: PropTypes.string,
-    params: PropTypes.object,
-    onDeleteClick: PropTypes.func,
-    onEditClick: PropTypes.func
-  }
+const ImportanceComponentWidget = (props) => {
+  const {
+    title,
+    params,
+    onDeleteClick,
+    onEditClick
+  } = props;
 
-  constructor(props) {
-    super(props);
-    this.title = props.title || 'Importance Component Widget';
-    this.params = props.params || {};
-    this.state = {
-      data: {
-        table_data: []
-      },
-      isLoading: true,
-      countSkips: 'No',
-    };
-  }
+  const [tableData, setTableData] = useState([]);
+  const [dataError, setDataError] = useState(true);
+  const [isLoading, setIsLoading] = useState(true);
+  const [countSkips, setCountSkips] = useState('No');
 
-  getData = () => {
-    this.setState({isLoading: true})
-    HttpClient.get([Settings.serverUrl, 'widget', 'importance-component'], this.params)
+  const getData = useCallback(() => {
+    setIsLoading(true);
+    HttpClient.get([Settings.serverUrl, 'widget', 'importance-component'], params)
       .then(response => {
         response = HttpClient.handleResponse(response, 'response');
         if (!response.ok) {
           throw Error(response.statusText);
         }
         return response.json();
       })
-      .then(data => this.setState({data: data, isLoading: false}))
+      .then(data => {
+        setTableData(data.table_data);
+        setIsLoading(false);
+        setDataError(false);
+        return data
```

Comment:
nitpick: No need to return data here, we have already used data to `setTableState`

Reply:
fixed, thanks for reviewing!